test/server.js

```diff
@@ -465,50 +465,44 @@ describe("server", function() {
                 }).then(function(response) {
                     var afterTimestamp = new Date();
                     assert.equal(helpers.getUpdateOneCallCount("notifications"), 2);
+                    // Verify that server has generated valid timestamp values
                     var firstTimestamp = helpers.getUpdateOneArgs("notifications", 0)[1].$set["data.since"];
                     var secondTimestamp = helpers.getUpdateOneArgs("notifications", 1)[1].$set["data.since"];
                     assert.isAtLeast(firstTimestamp.getTime(), beforeTimestamp.getTime());
                     assert.isAtMost(firstTimestamp.getTime(), afterTimestamp.getTime());
                     assert.equal(firstTimestamp, secondTimestamp);
-                    assert.deepEqual(helpers.getUpdateOneArgs("notifications", 0), [
-                        {
+                    // Verify that notification upserts have been performed with correct values
```

Comment:
That’s fine for now, but it would be a good idea to refactor and split the function soon as well. Every place where you’ve put a clarification comment might be a place that could become a new function. They wouldn’t be functions that you are expecting to reuse, but each function name could be a clear step within this test function.